pangeo_forge_orchestrator/routers/github_app.py

```diff
@@ -180,6 +198,63 @@ def get_storage_subpath_identifier(feedstock_spec: str, recipe_run: SQLModel):
     return f"{prefix}/{recipe_run.recipe_id}.{recipe_run.dataset_type}"
 
 
+def get_recipe_run_from_db(
+    *,
+    db_session: Session,
+    bakery_id: Optional[str] = None,
+    feedstock_id: Optional[str] = None,
+    head_sha: Optional[str] = None,
+    recipe_id: Optional[str] = None,
+    id: Optional[int] = None,
+):
+    """Check if a recipe with the given head_sha and feedstock_id exists in the database."""
+
+    if id:
+        statement = select(MODELS["recipe_run"].table).where(MODELS["recipe_run"].table.id == id)
+    else:
+        statement = select(MODELS["recipe_run"].table).where(
+            MODELS["recipe_run"].table.head_sha == head_sha,
+            MODELS["recipe_run"].table.recipe_id == recipe_id,
+            # TODO: should we be checking the bakery_id  and feedstock_id here?
+            # MODELS["recipe_run"].table.bakery_id == bakery_id,
+            # MODELS["recipe_run"].table.feedstock_id == feedstock_id,
```

Comment:
@cisaacstern, i noticed that we are currently using `recipe_id` and `sha` to retrieve a recipe run from the database? are these values enough? wouldn't this break if we had more than one bakeries?

https://github.com/pangeo-forge/pangeo-forge-orchestrator/blob/fb5caaf6994d1b73004b447ab00e5687ab941f2e/pangeo_forge_orchestrator/routers/github_app.py#L431-L434

Reply:
That's a great point. We should add `...table.bakery_id` to this.